src/olympia/blocklist/mlbf.py

```diff
@@ -72,178 +74,210 @@ def fetch_all_versions_from_db(excluding_version_ids=None):
     qs = Version.unfiltered.exclude(id__in=excluding_version_ids or ()).values_list(
         'addon__addonguid__guid', 'version'
     )
-    return list(qs)
+    return set(qs)
```

Comment:
It's actually fine for both to be sets. We need them to be sets because we do uniqueness subtraction and they must be iterable. We dont need the dict return in `fetch_all_version_from_db` as we can compose the data structure in the caller.